File: backend/src/main/java/edu/cit/olodin/feature/auth/AuthResponse.java
package edu.cit.olodin.feature.auth;

import lombok.Getter;

@Getter
public class AuthResponse {
    private String token;
    private String role;
    private boolean isNewUser;

    public AuthResponse(String token, String role, boolean isNewUser) {
        this.token = token;
        this.role = role;
        this.isNewUser = isNewUser;
    }
}

File: NearU_Mobile/app/src/main/java/edu/cit/olodin/nearu/mobile/model/AuthResponse.kt
package edu.cit.olodin.nearu.mobile.model

data class AuthResponse(
    val token: String
)

data class AuthData(
    val accessToken: String,
    val refreshToken: String,
    val user: UserDto
)

data class UserDto(
    val email: String,
    val firstname: String,
    val lastname: String,
    val role: String
)

data class ErrorResponse(
    val code: String,
    val message: String
)